test/functional/p2p-fullblocktest.py

```diff
@@ -291,7 +291,21 @@ def run_test(self):
         self.sync_blocks([b24], False, 16, b'bad-blk-length', reconnect=True)
 
         b25 = self.next_block(25, spend=out[7])
-        self.sync_blocks([b25], False)
+        # We've disabled this one in Dash as it blocks forever. This is because Dash handles too large blocks slightly
+        # different. Dash has 2 places where it checks the block size:
+        # 1. CheckBlock: This does NOT take the current height and DIP1 activation into account, which leads to to blocks
+        #    being accepted if size > 1000000
+        # 2. ConnectBlock: This takes DIP1 activation into account and leads to the same block from 1. being rejected
+        #
+        # In Dash CheckBlock passes, which then allows to call AcceptBlock and ConnectBlock, where the block then gets
+        # the BLOCK_FAILED_VALID status. This later leads to bad-prevblk when child blocks are announced via
+        # header. This in turn results in the node never sending getdata for b25, which causes us to block forever.
+        #
+        # Bitcoin bails out in ProcessNewBlock due to CheckBlock failing. This will then lead to CBlockIndex* never
+        # being updated with BLOCK_FAILED_VALID. When a child block is then announced to the node, it thinks that it
+        # never processed the invalid block and re-requests it together with the child block, which then makes the below
+        # test code pass.
+        # self.sync_blocks([b25], False)
```

Comment:
I don't like just commenting out code, but it makes sense in this case to avoid conflicts/make them easier to resolve when backporting.

Reply:
Yeah that was the intention. I know from experience that one of the hardest parts of backporting is to figure out why some change is missing or something is different. A comment helps later.